upload/upload.go

```diff
@@ -191,6 +195,19 @@ func NewHandler(
 
 		validator.Validate(vr)
 
-		w.WriteHeader(http.StatusAccepted)
+		response := responseBody{vr.RequestID}
+		jsonBody, err := json.Marshal(response)
+		if err != nil {
+			logerr("Unable to marshal JSON response body", err)
```

Comment:
In this case, you can't safely use `jsonBody` later.  Something pretty bad has likely happened, and you should return a 500 and short circuit.

Reply:
Added an internal server error and a return to handle that possibility